index.js

```diff
@@ -32,20 +34,29 @@ const snap = async (content, opts = {}) => {
     }
     if (opts.selector) {
       await page.waitFor(opts.selector, {timeout: 10 * 1000})
+    
+      source = await page.$(opts.selector)
+    } else {
+      if (!opts.clip) {
+        let view = await page.viewport()
```

Comment:
I'm not actually sure if this is necessary - don't know why the original code set a clip to the full viewport size (which seems to be the default).